geomstats/geometry/pullback_metric.py

```diff
@@ -103,6 +103,415 @@ def pickable_inner_product(i, j):
             pickable_inner_product(i, j)
             for i, j in itertools.product(range(self.dim), range(self.dim))
         )
-
         metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
         return metric_mat[0] if base_point.ndim == 1 else metric_mat
+
+
+class PullbackDiffeoMetric(RiemannianMetric, abc.ABC):
+    """
+    Pullback metric via a diffeomorphism.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension.
+    """
+
+    def __init__(self, dim, shape):
+        super(PullbackDiffeoMetric, self).__init__(dim=dim, shape=shape)
+
+        # Inner variable
+        self._embedding_metric = None
+        self._jacobian_diffeomorphism = None
+        self._inverse_jacobian_diffeomorphism = None
+
+        self.shape_dim = int(gs.prod(gs.array(shape)))
+        self.embedding_space_shape_dim = int(
+            gs.prod(gs.array(self.embedding_metric.shape))
+        )
+
+    @abc.abstractmethod
+    def create_embedding_metric(self):
+        r"""Create the metric this metric is in diffeomorphism with.
+
+        This instantiate the metric to use as image space of the
+        diffeomorphism.
+
+        -------
+        embedding_metric : RiemannianMetric object
+            The metric of the embedding space
+        """
+
+    @property
+    def embedding_metric(self):
+        r"""Property wrapper around the metric.
+
+        -------
+        embedding_metric : RiemannianMetric object
+            The metric of the embedding space
+        """
+        if self._embedding_metric is None:
```

Comment:
No, because, i want to do it only once when it is the first call and not at every call. Another way could be to drop the property mechanism and set `embedding_metric` using `self.create_embedding_metric` in `__init__` but this way ensure all the init is done before the call of `create_embedding_metric` which is more user-friendly for inherited class coding.

However I am not a python expert at all and I do not know the exact best practice

Reply:
You're right, I forgot the assignment part:

```
self._embedding_metric = self._embedding_metric or self.create_embedding_metric()
return self._embedding_metric
```

It is OK if you keep it as it is though (the `or` behavior may be harder to understand).

I think the property mechanism is perfect for this use case (@nguigs is also using it for `metric` in other parts of the code).